woocommerce/woocommerce-android
# Issue/5278 order creation product search

Summary
==========
Fixes issue #5278 by introducing the Search bar feature for the Product selection view using a very similar approach to how `ProductListFragment` and `ProductListViewModel` search works, since we're reusing the same List Adapter from there.

Screenshots
==========
| Product selection view with the search action icon  | Search bar expanded after clicking the search icon | Product list showing the result of the search |
| ------------- | ------------- | ------------- |
| ![Screenshot_20220114_042326](https://user-images.githubusercontent.com/5920403/149468201-e1c21823-da62-4dd0-9b9e-38931ed5ecea.png) | ![Screenshot_20220114_042337](https://user-images.githubusercontent.com/5920403/149468217-7f4bada5-b754-492c-87d2-0467c21eeb12.png) | ![Screenshot_20220114_042350](https://user-images.githubusercontent.com/5920403/149468232-ebdd04b5-bc9d-4a53-8bdd-e411b95e89e0.png) |

How to Test
==========
1. Make sure the order creation beta flag is enabled.
2. Open Order's list, and click on the FAB button.
3. Click on Add Product.
4. Verify that the Add Product view is presented with the correct Product list from the given store and the search menu icon displayed
5. Click on the menu icon and verify that the search bar expands inside the Toolbar
6. Place a query and verify the result shows up as expected
7. Click on any Product from the search result and verify if the view returns to the Order Creation form with the expected product added, if the Product is a Variable one, verify if the Variation list shows up instead.


Update release notes:
